From 78d4ad182923a7d4f70b0d339badfdac29e13d85 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 18 Aug 2024 16:31:34 +0300
Subject: [PATCH] [SW-164197] g3: set flash protected size to 256kb

In some flash devices the minimal protection granularity is 256kb.
For that reason the protected area is set to 256kb.
Until now the baremetal project was setting the protection for the whole
flash, that is not the desired configuration.

Change-Id: I2b046dea844cda0f989ff121add34ea8d156ac21
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.h b/arch/gaudi3/arc/common.h
index 6b02292..ba2e629 100644
--- a/arch/gaudi3/arc/common.h
+++ b/arch/gaudi3/arc/common.h
@@ -135,4 +135,16 @@
  */
 #define SPI_RX_POLL_TIMEOUT_US		2560 /* timeout in us */
 
+#ifndef __ASSEMBLY__
+#include <spi_flash.h>
+
+/* Let this file be included in any order and override the PROT_MASK */
+#ifdef PROT_MASK
+#undef PROT_MASK
+#endif
+/*  for macronix flash: BP0 | BP1 == 0x3 - means 4 blocks - up to 256KB (eROM) */
+#define PROT_MASK	(READ_STATUS_BP0 | READ_STATUS_BP1 | READ_STATUS_SRWD)
+
+#endif /* __ASSEMBLY__ */
+
 #endif /* COMMON_GAUDI3_ARC_GNU_H_ */
